web/templates/user_profile_modal.hbs

```diff
@@ -131,8 +131,35 @@
 
                     <div class="tabcontent" id="user-profile-groups-tab">
                         <div class="alert user-profile-group-list-alert"></div>
+                        {{#if show_user_group_container}}
+                            <div class="header-section">
+                                <h3 class="stream-tab-element-header">{{t 'Add {full_name} to groups'}}</h3>
+                            </div>
+                            <div id="groups-to-add">
+                                <div id="user-group-to-add">
+                                    <div class="add-user-group-container">
+                                        <div class="pill-container">
+                                            <div class="input" contenteditable="true"
+                                              data-placeholder="{{t 'Add user groups' }}">
+                                                {{~! Squash whitespace so that placeholder is displayed when empty. ~}}
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                                <div class="add-groups-button-wrapper">
+                                    <button type="button" name="subscribe" class="add-groups-button button small rounded">
+                                        {{t 'Add' }}
+                                    </button>
+                                </div>
+                            </div>
+                        {{/if}}
+                        <div class="stream-list-top-section">
+                            <div class="header-section">
+                                <h3 class="stream-tab-element-header">{{t 'Group membership' }}</h3>
+                            </div>
+                        </div>
                         <div class="subscription-group-list empty-list">
-                            <table class="user-group-list" data-empty="{{t 'No user group subscriptions.'}}"></table>
+                            <table class="user-group-list" data-empty="{{t 'Not a member of any groups.'}}" data-search-results-empty="{{t 'No matching user groups' }}"></table>
```

Comment:
This should be in the commit which adds the search input and probably a separate commit to improve the `data-empty` value.